library/src/armory/model/object_detection/yolov5_object_detector.py

```diff
@@ -74,23 +77,45 @@ def __init__(
             predictions_accessor=(
                 predictions_accessor or BoundingBoxes.as_torch(format=BBoxFormat.XYXY)
             ),
+            iou_threshold=iou_threshold,
+            score_threshold=score_threshold,
         )
 
         from yolov5.utils.general import non_max_suppression
         from yolov5.utils.loss import ComputeLoss
 
-        self.compute_loss = ComputeLoss(self._model.model.model)
+        self._detection_model = self._get_detection_model(self._model)
+        self.compute_loss = ComputeLoss(self._detection_model)
+
+        kwargs.update(
+            {
+                **({"conf_thres": score_threshold} if score_threshold else {}),
+                **({"iou_thres": iou_threshold} if iou_threshold else {}),
+            }
+        )
```

Comment:
Consider the following simpler implementation:
```suggestion
        if score_threshold is not None:
            kwargs["conf_thres"] = score_threshold
        if iou_threshold is not None:
            kwargs["iou_thres"] = iou_threshold
```